src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,198 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = (Node<K,V>[])new MyHashMap.Node[DEFAULT_INITIAL_CAPACITY];
+        this.threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        if (key == null) {
+            whenKeyNull(value);
+        } else if (nodeBusyAndKeysEquals(key)) {
+            table[bucketIndex(key)].value = value;
+        } else if (nodeBusyAndKeysDifferent(key)) {
+            if (noCollision(key)) {
+                table[bucketIndex(key)].next = addNode(key,value);
+                size++;
+            } else {
+                putWithCollision(key,value);
+            }
+        } else {
+            table[bucketIndex(key)] = addNode(key,value);
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K,V> node = table[bucketIndex(key)];
+        if (node != null) {
+            if (keysEquals(node,key)) {
+                return node.value;
+            } else {
+                return searchForCollision(node, key);
+            }
+        }
         return null;
     }
 
+    private void resize() {
+        size = 0;
+        Node<K,V>[] oldTable = table;
+        table = (Node<K,V>[])new MyHashMap.Node[table.length * 2];
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+        for (Node<K,V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private boolean nodeBusyAndKeysEquals(K key) {
+        return table[bucketIndex(key)] != null
+                && table[bucketIndex(key)].key.equals(key);
+    }
+
+    private boolean keysEquals(Node<K,V> node, K key) {
+        return (node.key == null && key == null) || node.key != null && node.key.equals(key);
+    }
+
+    private boolean nodeBusyAndKeysDifferent(K key) {
+        return table[bucketIndex(key)] != null
+                && !(table[bucketIndex(key)].key.equals(key));
+    }
```

Comment:
give short meaningful names, there is no need to copy-paste the whole condition.
```suggestion
    private boolean nodeBusyAndKeysEquals(K key) {
        return table[bucketIndex(key)] != null
                && table[bucketIndex(key)].key.equals(key);
    }

    private boolean keysEquals(Node<K,V> node, K key) {
        return (node.key == null && key == null) || node.key != null && node.key.equals(key);
    }

    private boolean nodeBusyAndKeysDifferent(K key) {
        return table[bucketIndex(key)] != null
                && !(table[bucketIndex(key)].key.equals(key));
    }
```

Reply:
> give short meaningful names, there is no need to copy-paste the whole condition.

about it, i dont have idea how change it in this context. Maybe u can explane, please